pkg/config/new.go

```diff
@@ -0,0 +1,124 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	cachedConfigError error
+	cachedConfigMutex sync.Mutex
+	cachedConfig      *Config
+)
+
+// Options to use when loading a Config via New().
+type Options struct {
+	// Set the loaded config as the default one which can later on be
+	// accessed via Default().
+	SetDefault bool
+
+	// Additional configs that will be loaded after all system/user configs
+	// and environment variables but the _OVERRIDE one.
+	additionalConfigs []string
+}
+
+// New returns a Config as described in the containers.conf(5) man page.
+func New(options *Options) (*Config, error) {
+	if options == nil {
+		options = &Options{}
+	}
+	cachedConfigMutex.Lock()
+	defer cachedConfigMutex.Unlock()
```

Comment:
Why should we have a lock for this?
If a caller wants to read all configs it should just do that without lock.

The reason we have a lock for Default() is because we store the result in the global `cachedConfig` var and when two caller call us at the same time we only read once and then return the cached result from there on.

If this is only for the Reload() case, I think we can it is better to special case  `SetDefault` and only lock in this case.

The reason is if we start reading modules with this function and then you use the API to create containers and parallel this mutex can be a bottleneck for no reason.

Reply:
That sounds good to me.

Reply:
This gave me an idea to speed up `Default()`.  We can avoid the unconditional lock.